app/api/v1/endpoints/prediction.py

```
@@ -1,45 +1,37 @@
+        predictions = await predict_pet_disease_custom_vision(request.image_url, request.pet_type)
+        return predictions
+    except ValueError as e:  # pet_type 오류 처리
+        raise HTTPException(status_code=400, detail=str(e))
+    except requests.exceptions.RequestException as e:
+        raise HTTPException(status_code=400, detail=f"Image download failed: {str(e)}")
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Prediction failed: {str(e)}")
 
-    # 예측 요청
+@router.post("/torch/predict", response_model=List[PredictionResult])
+async def predict_image(request: ImagePredictionRequest):
```

Ruff (0.8.2)

Redefinition of unused `predict_image` from line 11

(F811)